### PR TITLE
Prepare backend for admin module

### LINKED ISSUE #602: Backend: Create projectType class
This class will contain name and a list of questions for a given project type

Needed in QuestionTemplate and Evaluation

### LINKED ISSUE #604: Backend: Mutation to duplicate question set from existing question set

### LINKED ISSUE #605: Backend: CreateEvaluation mutation needs project type as argument

### LINKED ISSUE #606: Backend: Query to ask for set of project types

### PR DESCRIPTION
**Pull request by ErlendHaa**
This PR introduces a new table "Sub Project" in our database. A sub project is essentially a set of QuestionTemplates. In the real world a subproject might be "GreenField" or "BrownField". There is many-to-many relationship between subproject and questionTemplates. I.e. a subproject contains many questionTemplates, but a questionTemplate can also used in multiple subprojects.

The subprojects are put to use in the CreateEvaluationDialog, where it's now mandatory to choose which sub project (questionnaire set) to use in the evaluation.

Additionally, several new mutations that will become useful in the new admin module is added. Authorisation mechanisms for the admin queries/mutations are not yet added

Closes #602 
Closes #604 
Closes #605
Closes #606

### CODE REVIEW

File: backend/api/Context/BmtDbContext.cs [hunk 1]
```diff
@@ -15,6 +15,7 @@ public class BmtDbContext : DbContext
         public DbSet<Note> Notes { get; set; }
         public DbSet<ClosingRemark> ClosingRemarks { get; set; }
         public DbSet<QuestionTemplate> QuestionTemplates { get; set; }
+        public DbSet<SubProject> SubProjects { get; set; }
```
Comment (petterwildhagen) [inline backend/api/Context/BmtDbContext.cs:18]: rename to ProjectType, ref https://github.com/equinor/fusion-bmt/issues/522. ?
Is SubProject a concept in BMT (in the domain)?

File: backend/api/Context/InitContent.cs [hunk 2]
```diff
@@ -251,6 +254,21 @@ private static List<Project> GetProjects()
             return projects;
         }
 
+        private static List<SubProject> GetSubProjects()
```
Comment (petterwildhagen) [inline backend/api/Context/InitContent.cs:257]: GetProjectTypes ?

**Review by petterwildhagen — COMMENTED** [review]
It seems a new domain concept 'Sub Projects' has been introduced. Replace with Project Type?

File: backend/api/Context/BmtDbContext.cs (hunk 1 above)
Comment (ErlendHaa) [reply backend/api/Context/BmtDbContext.cs]: The gurus have spoken, the new name will be `ProjectCategory`

File: backend/api/GQL/Mutation.cs [hunk 3]
```diff
@@ -43,27 +50,45 @@ ILogger<Mutation> logger
...
+                name:                 name,
+                project:              project,
+                previousEvaluationId: previousEvaluationId
+            );
+
+            _participantService.Create(
+                azureUniqueId: azureUniqueId,
+                evaluation:    evaluation,
+                organization:  Organization.All,
+                role:          Role.Facilitator
```
Comment (ReedOnly) [inline backend/api/GQL/Mutation.cs:83]: Typed parameters for readability?

File: backend/tests/GQL/Mutation.cs [hunk 4]
```diff
@@ -64,7 +71,8 @@ public MutationTest()
         protected Evaluation CreateEvaluation(
             string name = null,
             string projectId = null,
-            string previousEvaluationId = null)
+            string previousEvaluationId = null,
+            string projectCategoryId = null)
```
Comment (ReedOnly) [inline backend/tests/GQL/Mutation.cs:75]: should be disable this Codacy pattern? Here is reasoning behind not using optional parameters: https://app.codacy.com/gh/equinor/fusion-bmt/pullRequest?prid=8068964

**Review by ReedOnly — COMMENTED** [review]
Looks very good. I really appreciate your focus on clarity in the code. It makes reading the code like poetry 🏆

File: backend/tests/GQL/Mutation.cs (hunk 4 above)
Comment (ErlendHaa) [reply backend/tests/GQL/Mutation.cs:75]: Nah, their argument is fair enough, especially for production code. But for test code I still think it's ok to use optional arguments if it gives any real benefit. In this case overloads aren't really feasible as the number of combinations becomes too great.

**Review by ReedOnly — APPROVED** [review]

**Pull request merged**